src/stack-unix/udpv4v6_socket.ml

```diff
@@ -191,6 +190,11 @@ let listen t ~port callback =
                            | Some v4 -> Ipaddr.V4 v4
                          in
                          let dst = Ipaddr.(V6 V6.unspecified) in (* TODO *)
+                         let buf =
+                           let b = Cstruct.create_unsafe len in
+                           Cstruct.blit buf 0 b 0 len;
+                           b
+                         in
```

Comment:
maybe use Cstruct.copy?

Reply:
Cstruct.copy is Cstruct.t -> int -> int -> string, that is, `Cstruct.copy t off len` is the same as `Cstruct.to_string t ~off ~len` :/ I didn't see a function to copy a cstruct (even though Cstruct.copy would be an obvious name for such a function)

Reply:
Hi @palainp, indeed, `Cstruct.sub_copy` does what I want, but it only exists in unreleased cstruct. I can put a comment in the code to use `Cstruct.sub_copy` in the future.

Reply:
Oh sorry, didn't checked the last release date :(

Reply:
I will cut a release of `cstruct` as soon as I can but some improvements are not merged yet.